### ISSUE TITLE
Remove from map

### ISSUE THREAD

**Issue by peter-alexander** [GuillaumeAmat/leaflet-overpass-layer#14]
Hey,

I'd like to remove the OverpassLayer from the map using the layer switching control.

This is what I tried:

    <script>
      var mapnik = L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png',{
                    subdomains:['a', 'b', 'c'],
                    attribution:'© <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'});

      var bicycleshop_icon = L.icon({
                              iconUrl:'https://wiki.openstreetmap.org/w/images/4/4e/Icon_bicycle_shop.png',
                              iconSize:[22, 19]});

      var bicycleshop = new L.OverPassLayer({'query': '(node({{bbox}})[shop=bicycle];);out qt;',
                             markerIcon: bicycleshop_icon,minZoom:11});

      var baseLayers = {
    		"OpenStreetMap Mapnik":mapnik,
      };

      var overlays = {
      		"bike shop":bicycleshop,
      };

      var map = L.map('map', {
                  center:[48.208, 16.37],
                  zoom:13,
                  layers:[mapnik],
      });

      L.control.layers(baseLayers, overlays, {collapsed:false}).addTo(map);

    </script>
  </body>
</html>
`

**Comment by GuillaumeAmat** [comment]
Hey,

What's your issue exactly? I'm not familiar with the *Layers Control* but as far as [I can](http://leafletjs.com/examples/layers-control/#layers-control) [see](http://leafletjs.com/reference.html#control-layers), your code seems fine.

**Comment by peter-alexander** [comment]
Adding the overpass layers to the map works fine but when removing they stay visible on the map instead of disappearing. Demo: [https://fahrrad.lima-city.de/overpass_remove_fail.html](url)

**Comment by GuillaumeAmat** [comment]
Hi!

Sorry for the delay, I'm really busy right now ;)
i just released the 2.6.0 version with the fix for your issue. Feel free to reopen it if you still have trouble with that.

**Comment by peter-alexander** [comment]
Hi!
Thank you for taking time trying to fix it.
The layer is somehow removed from the map but there is an error. The result is that it's not possible to readd the layer after it was removed. I updated the link above with the new version. Because I need to use a hash I also created another one using hash. The layers are not removed from the hash (it's working with non-opverpass layers): [https://fahrrad.lima-city.de/overpass_remove_fail_w_hash.html](https://fahrrad.lima-city.de/overpass_remove_fail_w_hash.html)

Error displayed in Firefox JS console on remove is "TypeError: t is undefined"
Error displayed in Chrome JS console on remove is "Uncaught TypeError: Cannot read property '_leaflet_id' of undefined"
